### PR TITLE
Return away_temperature if eco_temperature is undefined

### PR DESCRIPTION
**Pull request by mriksman**
On the UK Thermostats at least, eco_temperature_low_c and eco_temperature_high_c are undefined when in Eco mode. The correct Eco temperatures are located in away_temperature_high_c and away_temperature_low_c.

### CODE REVIEW

File: lib/nest-thermostat-accessory.js [hunk 1]
```diff
@@ -238,7 +238,7 @@ NestThermostatAccessory.prototype.getTargetTemperature = function () {
 NestThermostatAccessory.prototype.getCoolingThresholdTemperature = function () {
   switch (this.device.hvac_mode) {
   case "eco":
-    return this.device.eco_temperature_high_c;
+    return this.device.eco_temperature_high_c ? this.device.eco_temperature_high_c : this.device.away_temperature_high_c;
```
Comment (chrisjshull) [inline lib/nest-thermostat-accessory.js]: Same thing, slightly simpler:
`return this.device.eco_temperature_high_c || this.device.away_temperature_high_c;`

Can you add an in-file comment explaining the need for this fallback?

File: lib/nest-thermostat-accessory.js [hunk 2]
```diff
@@ -248,7 +248,7 @@ NestThermostatAccessory.prototype.getCoolingThresholdTemperature = function () {
 NestThermostatAccessory.prototype.getHeatingThresholdTemperature = function () {
   switch (this.device.hvac_mode) {
   case "eco":
-    return this.device.eco_temperature_low_c;
+    return this.device.eco_temperature_low_c ? this.device.eco_temperature_low_c : this.device.away_temperature_low_c;
```
Comment (chrisjshull) [inline lib/nest-thermostat-accessory.js]: ditto

**Comment by mriksman** [comment]
Hmmm. I've never had to do a 'Change Requested' before. I'm pretty new to github. I made the change, commit, push. Not sure if that's it?

**Comment by mriksman** [comment]
This will conflict with the changes you made in 2.1.2 eh?

**Comment by chrisjshull** [comment]
yeah, I merged it though

**Comment by chrisjshull** [comment]
Thanks!

**Comment by mriksman** [comment]
No problem. Glad you know what you're doing!

**Comment by mriksman** [comment]
I wonder why my thermostat/API isn't using eco_thermostat?
https://developers.nest.com/documentation/cloud/api-thermostat#away_temperature_high_c_deprecated

**Comment by mriksman** [comment]
Sh*t. I needed to update my permissions on the API. I first set up Nest in 2016 where the Thermostat was v5. eco_thermostat is in v6. I updated the permissions, reauthorised, and now eco_thermostat exists. 
So this 'fix' isn't about UK thermostats. 
It's about those idiots (like me) who were on deprecated API permissions.

**Comment by chrisjshull** [comment]
ah, good to know. How about a PR that updates the comments in the code to note that?

**Pull request merged**